### PR TITLE
New Changeset format

### PR DESCRIPTION
**Pull request by yann-achard-MS**
## Description

This PR updates the Changeset format to reflect more recent design considerations. The timing for this PR is motivated by the need to have *something* to work with in order to make progress on prototyping other parts of the system.

### CODE REVIEW

File: packages/dds/tree/src/changeset/format.ts [hunk 1]
```diff
@@ -729,35 +318,98 @@ export interface HasLength {
...
 export type TreePath = TreeChildPath | TreeRootPath;
 
+export enum RangeType {
+	Set = "Set",
+	Slice = "Slice",
+}
 /**
  * The relative location of the sibling based on which a segment or segment boundary is defined.
  */
 export enum Sibling {
```
Comment (alex-pardes) [inline packages/dds/tree/src/changeset/format.ts:336]: Should this be removed?

File: packages/dds/tree/src/changeset/format.ts [hunk 2]
```diff
@@ -729,35 +318,98 @@ export interface HasLength {
...
-    Next,
+	/**
+	 * Used for, e.g., insertion after a given node.
+	 */
+	Prev,
+	/**
+	 * Used for, e.g., insertion before a given node.
+	 */
+	Next,
+}
```
Comment (alex-pardes) [inline packages/dds/tree/src/changeset/format.ts:345]: Consider removing

File: packages/dds/tree/src/changeset/format.ts (hunk 1 above)
Comment (yann-achard-MS) [reply packages/dds/tree/src/changeset/format.ts]: Removed indeed.

File: packages/dds/tree/src/changeset/format.ts (hunk 2 above)
Comment (yann-achard-MS) [reply packages/dds/tree/src/changeset/format.ts]: Removed.

**Review by alex-pardes — APPROVED** [review]

**Pull request merged**